test/dlc_tests/sanity/test_security_check.py

```diff
@@ -54,6 +66,297 @@ def run_scan(ecr_client, image):
+        image, ecr_image_vulnerability_list, "current-ecr-scanlist"
+    )
+    original_filepath_for_allowlist = get_ecr_scan_allowlist_path(image)
+    edited_files = [{"s3_filename": s3_filename_for_allowlist, "github_filepath": original_filepath_for_allowlist}]
+    vulnerabilities_fixable_by_upgrade = get_vulnerabilites_fixable_by_upgrade(
+        image_allowlist, ecr_image_vulnerability_list, upgraded_image_vulnerability_list
+    )
+    newly_found_non_fixable_vulnerabilites = upgraded_image_vulnerability_list - image_allowlist
+    fixable_list = []
+    if vulnerabilities_fixable_by_upgrade:
+        fixable_list = vulnerabilities_fixable_by_upgrade.vulnerability_list
+    s3_filename_for_apt_upgrade_list = s3_filename_for_allowlist.replace("allowlist.json", "apt-upgrade-list.txt")
+    original_filepath_for_apt_upgrade_list = "/".join(
+        original_filepath_for_allowlist.split("/")[:-1] + ["apt-upgrade-list.txt"]
+    )
+    new_package_list = fixable_list if isinstance(fixable_list, list) else list(fixable_list.keys())
+    create_and_save_package_list_to_s3(
+        original_filepath_for_apt_upgrade_list, new_package_list, s3_filename_for_apt_upgrade_list
+    )
+    edited_files.append(
+        {"s3_filename": s3_filename_for_apt_upgrade_list, "github_filepath": original_filepath_for_apt_upgrade_list}
+    )
+    newly_found_non_fixable_list = []
+    if newly_found_non_fixable_vulnerabilites:
+        newly_found_non_fixable_list = newly_found_non_fixable_vulnerabilites.vulnerability_list
+    message_body = {
+        "edited_files": edited_files,
+        "fixable_vulnerabilities": fixable_list,
+        "non_fixable_vulnerabilities": newly_found_non_fixable_list,
+    }
+    invoke_lambda(function_name="trshanta-ECR-AS", payload_dict=message_body)
+    return_dict = deepcopy(message_body)
+    return_dict["s3_filename_for_allowlist"] = s3_filename_for_allowlist
+    return_dict["s3_filename_for_current_image_ecr_scan_list"] = s3_filename_for_current_image_ecr_scan_list
+    return return_dict
+
+
+def is_image_covered_by_allowlist_feature(image):
+    """
+    This method checks if the allowlist feature has been enabled for the image
+
+    :param image: str, Image URI
+    """
+    if any(substring in image for substring in ["example","neuron"]):
+        return False
+    for framework in LOWER_THRESHOLD_IMAGES.keys():
+        if framework in image:
+            if any(version in image for version in LOWER_THRESHOLD_IMAGES[framework]):
+                return True
+            return False
+    return False
+
+
+def set_minimum_threshold_level(image):
+    """
+    This method sets the value for MINIMUM_SEV_THRESHOLD
+
+    :param image: str Image URI for which threshold has to be set
+    """
+    global MINIMUM_SEV_THRESHOLD
+    if is_image_covered_by_allowlist_feature(image):
+        MINIMUM_SEV_THRESHOLD = "MEDIUM"
+        return
+    MINIMUM_SEV_THRESHOLD = "HIGH"
```

Comment:
How do global variables work in the context of multithreaded pytest executions for multiple images? Are the global variables within each script managed separately for each thread?

Reply:
I read that `pytest -n auto` uses multiprocessing to run the tests. In that case, change to Global variables in one process is not reflected in the other. However, to be on the safer side, I have removed the usage of global variables altogether.